src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -140,6 +144,62 @@ public synchronized void close() throws IOException {
             return;
         }
 
+        try {
+            stopGracefully();
+        } finally {
+            stopForcefully();
+        }
+    }
+
+    private void stopGracefully() {
+        int pid = Processes.newPidProcess(process).getPid();
+        // Global lock since we're messing with a shared resource (the console)
+        boolean sentConsoleCtrlEvent = false;
+        synchronized (Kernel32Ex.INSTANCE) {
+            Kernel32 k32 = Kernel32.INSTANCE;
+            if (!k32.AttachConsole(pid)) {
+                logger.error("AttachConsole error {}", k32.GetLastError());
+                // Console already attached so we cannot signal it
+                if (k32.GetLastError() == ERROR_ACCESS_DENIED) {
+                    logger.info("AttachConsole failed for PID: {}, calling FreeConsole on it", pid);
+                    // Oddly, calling FreeConsole seems to make subsequent calls work?
+                    if (!k32.FreeConsole()) {
+                        logger.error("FreeConsole error {}", k32.GetLastError());
+                    }
+                }
+                return;
+            }
+            Kernel32Ex k32Ex = Kernel32Ex.INSTANCE;
+            try {
+                if (!k32Ex.SetConsoleCtrlHandler(null, true)) {
+                    logger.error("SetConsoleCtrlHandler add error {}", k32.GetLastError());
+                    return;
+                }
+                if (k32.GenerateConsoleCtrlEvent(0, 0)) {
+                    sentConsoleCtrlEvent = true;
+                } else {
+                    logger.error("GenerateConsoleCtrlEvent error {}", k32.GetLastError());
+                }
+            } finally {
+                if (!k32.FreeConsole()) {
+                    logger.error("FreeConsole error {}", k32.GetLastError());
+                }
+                if (!k32Ex.SetConsoleCtrlHandler(null, false)) {
+                    logger.error("SetConsoleCtrlHandler remove error {}", k32.GetLastError());
+                }
+            }
+        }
+
+        try {
+            if (sentConsoleCtrlEvent) {
+                process.waitFor(gracefulShutdownTimeout, TimeUnit.SECONDS);
+                logger.info("Process stopped gracefully: {}", pid);
```

Comment:
is this what we log for Unix? I don't want us to have extraneous logs

Reply:
this should be debug at highest

Reply:
changed to debug!